### PR TITLE
Move thrift middlewares into thriftbp package

### PR DESCRIPTION
**Pull request by pacejackson**

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -2,6 +2,7 @@ github.com/BurntSushi/toml v0.3.1 h1:WXkYYl6Yr3qBf1K79EBnL4mak0OimBfB0XUf9Vl28OQ
 github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
 github.com/VividCortex/gohistogram v1.0.0 h1:6+hBz+qvs0JOrrNhhmR7lFxo5sINxBCGXrdtl/UvroE=
 github.com/VividCortex/gohistogram v1.0.0/go.mod h1:Pf5mBqqDxYaXu3hDrrU+w6nw50o/4+TcAqDqk/vUH7g=
+github.com/apache/thrift v0.13.0 h1:5hryIiq9gtn+MiLVn0wP37kb/uTeRZgN08WoCsAhIhI=
```
Comment (fishy) [inline go.sum:5]: this line gets removed by `go mod tidy`.

File: httpbp/middlewares.go [hunk 2]
```diff
@@ -127,14 +126,20 @@ func InitializeEdgeContextFromTrustedRequest(
...
-		return edgecontext.InitializeEdgeContext(ctx, impl, logger, factory)
+	if !truster.TrustEdgeContext(r) {
+		return ctx
 	}
-	return ctx
+
+	header := r.Header.Get(EdgeContextHeader)
+	ec, err := edgecontext.FromHeader(header, impl)
+	if err != nil {
+		log.Error("Error while parsing EdgeRequestContext: " + err.Error())
```
Comment (fishy) [inline httpbp/middlewares.go:138]: this is the zap logger? then prefer this instead:
```
log.Errorw("Error while parsing EdgeRequestContext", "err", err)
```

File: thriftbp/tracing.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+// then use that span and the parent context object with this call,
+// then use the returned context object in the thrift call.
+// Something like:
+//
+//     span, clientCtx := opentracing.StartSpanFromContext(
+//       ctx,
+//       "myCall",
+//       tracing.SpanTypeOption{Type: tracing.SpanTypeClient},
+//     )
+//     result, err := client.MyCall(clientCtx, arg1, arg2)
```
Comment (fishy) [inline thriftbp/tracing.go:28]: There's one step missing before this line:

```
clientCtx = thriftbp.CreateThriftContextFromSpan(clientCtx, traceing.AsSpan(span))
```

Or maybe inline that into the client call.

File: thriftbp/tracing.go [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+//     // Or: span.Stop(ctx, err)
```
Comment (fishy) [inline thriftbp/tracing.go:29]: this also cannot be used directly and needs `tracing.AsSpan(span).Stop(ctx, err)`. Maybe remove this line and recommend using opentracing api instead.

File: thriftbp/middlewares.go [hunk 5]
```diff
@@ -63,8 +67,8 @@ var (
 //		...
 //		N. Middlewares[n]
 //
-// It is reccomended that you pass in tracing.InjectThriftServerSpan and the
-// Middleware returned by edgecontext.InjectThriftEdgeContext as the first two
+// It is reccomended that you pass in tracing.InjectServerSpan and the
```
Comment (konradreiche) [inline thriftbp/middlewares.go:70]: :no_bell: s/reccomended/recommend

**Pull request merged**